e2e/lvm/pvc_test.go

```diff
@@ -15,75 +16,198 @@ import (
 const (
 	timeout  = time.Minute * 5
 	interval = time.Second * 30
+	size     = "5Gi"
 )
 
-// Test to verify the PVC status
 func PVCTest() {
 
 	Describe("PVC Tests", func() {
-		var filepvc *k8sv1.PersistentVolumeClaim
-		var filepod *k8sv1.Pod
-		var blockpvc *k8sv1.PersistentVolumeClaim
-		var blockpod *k8sv1.Pod
+		var pvc *k8sv1.PersistentVolumeClaim
+		var pod *k8sv1.Pod
+		var snapshot *snapapi.VolumeSnapshot
+		var clonePvc *k8sv1.PersistentVolumeClaim
+		var clonePod *k8sv1.Pod
+		var restorePvc *k8sv1.PersistentVolumeClaim
+		var restorePod *k8sv1.Pod
+		test := tests.DeployManagerObj.GetCrClient()
+
+		Context("create pvc, pod, snapshots, clones", func() {
+			It("Tests PVC operations for VolumeMode=file system", func() {
+				By("Creating pvc and pod")
+				pvc = tests.GetSamplePVC(tests.StorageClass, size, "lvmfilepvc", k8sv1.PersistentVolumeFilesystem)
+				pod = tests.GetSamplePod("lvmfilepod", pvc.Name)
+				err := test.Create(context.TODO(), pvc)
+				Expect(err).To(BeNil())
+				err = test.Create(context.TODO(), pod)
+				Expect(err).To(BeNil())
+
+				By("Verifying that the PVC(file system) is bound and the Pod is running")
+				Eventually(func() bool {
+					err := test.Get(context.TODO(), types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, pvc)
+					return err == nil && pvc.Status.Phase == k8sv1.ClaimBound
+				}, timeout, interval).Should(BeTrue())
+				fmt.Printf("PVC %s is bound\n", pvc.Name)
+
+				Eventually(func() bool {
+					err = test.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, pod)
+					return err == nil && pod.Status.Phase == k8sv1.PodRunning
+				}, timeout, interval).Should(BeTrue())
+				fmt.Printf("Pod %s is running\n", pod.Name)
+
+				By("Creating a Snapshot of the file-pvc")
+				snapshot = tests.GetSampleVolumeSnapshot(pvc.Name, tests.StorageClass)
+				err = test.Create(context.TODO(), snapshot)
+				Expect(err).To(BeNil())
+				fmt.Printf("Snapshot %s is created\n", snapshot.Name)
+
+				By("Creating a clone of the file-pvc")
+				clonePvc = tests.GetPvc(size, pvc.Name, k8sv1.PersistentVolumeFilesystem, tests.StorageClass, "clone", "PersistentVolumeClaim")
```

Comment:
Any input should be in terms of the PVC being cloned. For example:
clonePvc = tests.GetPvc(pvc.size, pvc.Name, pvc.AccessMode, tests.StorageClass, "clone")

The last argument can be removed.
If it is a clone, it is of type PVC.
If it is a snapshot, it is of type VolumeSnapshot.